pyiron_base/job/util.py

```diff
@@ -393,7 +394,8 @@ def _job_store_before_copy(job):
         delete_file_after_copy = True
     else:
         delete_file_after_copy = False
-    job.to_hdf()
+    if not copy_exists:
+        job.to_hdf()
```

Comment:
There's an issue, where someone might copy a job, modify it and then copy it again.  If I understand this correctly this would not be caught by this solution.

Don't have a better idea just now, though.

Reply:
I also share this concern. Instead of the flag I'd like to try loading an instance of the job that's about to get saved and compare the pre-save, the go ahead and save if either the load failed or the two instances differ. However, I don't think we have such an equality comparison....

Reply:
@pmrv In that case, you are absolutely right. But as @liamhuber said, unless there is another method that compares if any dataset in the job has been modified, I also can't think of a better way. I can foresee this part resulting in a larger issue.

Reply:
The only thing I worry about in fixing it later is that right now the code works slow but predictably; after these changes it will work fast but dangerously. (I realize I am very lazy with language here and (slow,fast)==(memory hog, memory efficient). In general I am much more comfortable with slow and predictable/safe.

Reply:
How about adding a tag in the `copy_to` method like `rewrite_hdf` which is by default `True`, and for use cases like mine, I can manually set it to `False`?

Reply:
This, plus @pmrv's suggestion on modifying and using `rewrite_hdf5` in `copy_to`.

Reply:
Yes, if the dangerous behaviour has an energy barrier then I'm satisfied.